be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/LaoValidator.scala

```diff
@@ -1,140 +1,328 @@
+          }
 
-            if (!validateTimestampStaleness(data.last_modified)) {
-              Right(validationError(s"stale 'last_modified' timestamp (${data.last_modified})"))
-            } else if (!validateWitnesses(data.witnesses)) {
-              Right(validationError("duplicate witnesses keys"))
-            } else if (expectedHash != data.id) {
-              Right(validationError("unexpected id"))
-            } else if (!validateChannelType(ObjectType.LAO, channel)) {
-              Right(validationError(s"trying to write an UpdateLao message on wrong type of channel $channel"))
-            } else {
-              Left(rpcMessage)
-            }
-          case Some(Failure(ex: DbActorNAckException)) => Right(PipelineError(ex.code, s"validateUpdateLao failed : ${ex.message}", rpcMessage.getId))
-          case reply                                   => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, s"validateUpdateLao failed : unexpected DbActor reply '$reply'", rpcMessage.getId))
-        }
-
-      case _ => Right(validationErrorNoMessage(rpcMessage.id))
+        case _ => Right(validationErrorNoMessage(rpcMessage.id))
+
+      }
     }
   }
+
+  /** Check if all witnesses are distinct
+    *
+    * @param rpcMessage
+    *   rpc message to validate
+    * @param witnesses
+    *   witnesses to check
+    * @param error
+    *   the error to forward in case the witnesses are not all distinct
+    * @return
+    *   GraphMessage: passes the rpcMessages to Left if successful right with pipeline error
+    */
+  private def checkWitnesses(rpcMessage: JsonRpcRequest, witnesses: List[PublicKey], error: PipelineError): GraphMessage = {
+    if (validateWitnesses(witnesses))
+      Left(rpcMessage)
+    else
+      Right(error)
+  }
+
+  /** Check for chan1 and chan2 equality
+    *
+    * @param rpcMessage
+    *   rpc message to validate
+    * @param chan1
+    *   the first channel to compare
+    * @param chan2
+    *   the second channel to compare
+    * @param error
+    *   the error to forward in case the channels are not equals
+    * @return
+    *   GraphMessage: passes the rpcMessages to Left if successful right with pipeline error
+    */
+  private def checkChannel(rpcMessage: JsonRpcRequest, chan1: Channel, chan2: Channel, error: PipelineError): GraphMessage = {
+    if (chan1 == chan2)
+      Left(rpcMessage)
+    else
+      Right(error)
+  }
+
+  /** Check if the LAO's name is empty
+    *
+    * @param rpcMessage
+    *   rpc message to validate
+    * @param name
+    *   the name to check
+    * @param error
+    *   the error to forward in case the name si empty
+    * @return
+    *   GraphMessage: passes the rpcMessages to Left if successful right with pipeline error
+    */
+  private def checkLAOName(rpcMessage: JsonRpcRequest, name: String, error: PipelineError): GraphMessage = {
+    if (name.nonEmpty)
+      Left(rpcMessage)
+    else
+      Right(error)
```

Comment:
Minor comment: for future implementation, it could be useful to generalize this function.
By generalizing, I simply mean changing slightly its signature for further use and putting it somewhere else.